src/rs/scramble/scramble_search.rs

```diff
@@ -56,17 +56,20 @@ pub(crate) fn basic_idfs(
 pub(crate) fn filtered_search(
     scramble_pattern: &KPattern,
     generators: Generators,
-    min_optimal_moves: Option<usize>,
-    min_scramble_moves: Option<usize>,
+    min_optimal_moves: usize,
+    min_scramble_moves: Option<usize>, // inclusive
+    max_scramble_moves: Option<usize>, // exclusive
 ) -> Option<Alg> {
+    assert_ne!(min_optimal_moves, 0); //<<<
```

Comment:
This should be good, we can figure out how to do exhaustive assertions inside the `.search(…)` function.